core/sql/src/main/java/org/trafodion/sql/udr/SimpleUDR.java

```diff
@@ -0,0 +1,105 @@
+// @@@ START COPYRIGHT @@@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+// @@@ END COPYRIGHT @@@
+
+package org.trafodion.sql.udr;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.trafodion.sql.udr.TypeInfo.SQLTypeCode;
+
+public abstract class SimpleUDR extends UDR {
+    private static Logger log = Logger.getLogger(SimpleUDR.class.getName());
+    private List<ColumnInfo> columns = new ArrayList<ColumnInfo>();
+    UDRInvocationInfo info = null;
```

Comment:
The UDR class can be used for multiple UDR invocations, possibly at the same time, so you can't cache the UDRInvocationInfo in it.